### PR TITLE
chore: extra cleanup after d2g migration

### PR DESCRIPTION
**Pull request by matt-boris**
Fixes https://github.com/taskcluster/taskcluster/issues/6380.

### CODE REVIEW

File: taskcluster/ci/go/kind.yml [hunk 1]
```diff
@@ -2,30 +2,71 @@
...
+        git --version
+        go version
+        go env
+        if [ ! -d taskcluster/.git ]; then
+          rm -rf taskcluster
+          git clone {head_repository} taskcluster
+        fi
+        cd taskcluster
+        git fetch {head_repository} "+${{GITHUB_SHA}}:refs/heads/X${{TASK_ID}}"
+        git checkout -f "X${{TASK_ID}}"
```
Comment (lotas) [inline taskcluster/ci/go/kind.yml:31]: just curious, do you remember how that `X{{TASK_ID}}` refs is created? Do we create a branch for each task, or previous `get fetch` is doing something I'm not aware of 🤔

File: tools/jsonschema2go/jsonschema.go [hunk 2]
```diff
@@ -661,7 +661,7 @@ func (subSchema *JsonSubSchema) link(job *Job) (err error) {
 	if ref := subSchema.Ref; ref != nil && *ref != "" {
 		subSchema.RefSubSchema = job.result.SchemaSet.all[subSchema.RefSchemaURL]
 		if subSchema.RefSubSchema == nil {
-			return fmt.Errorf("Subschema %v not loaded when updating %v", subSchema.RefSchemaURL, subSchema.SourceURL)
+			return fmt.Errorf("subschema %v not loaded when updating %v", subSchema.RefSchemaURL, subSchema.SourceURL)
```
Comment (lotas) [inline tools/jsonschema2go/jsonschema.go:664]: u no like uppercase? :)

Comment (lotas) [reply tools/jsonschema2go/jsonschema.go:664]: let's hope no-one is doing case-sensitive checks for those errors 😬

**Review by lotas — APPROVED** [review]
nice!

File: taskcluster/ci/go/kind.yml (hunk 1 above)
Comment (matt-boris) [reply taskcluster/ci/go/kind.yml:31]: I'm not sure.. this is how other tasks do it as well so I just went along with the magic 😂

I think a new branch is created for each task though, yes.

File: tools/jsonschema2go/jsonschema.go (hunk 2 above)
Comment (matt-boris) [reply tools/jsonschema2go/jsonschema.go:664]: Haha, it fails the staticcheck :)

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Comment (lotas) [reply tools/jsonschema2go/jsonschema.go:664]: good to know, thanks

File: taskcluster/ci/go/kind.yml (hunk 1 above)
Comment (lotas) [reply taskcluster/ci/go/kind.yml:31]: aaah.. that's the syntax to checkout specific SHA, yet create a branch (local ref) with this name.. neat

**Pull request merged**